src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3753,8 +3800,8 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 );
                 throw SemanticAbort();
             }
-            ASR::expr_t *index = nullptr;
             if (ASR::is_a<ASR::Dict_t>(*type)) {
```

Comment:
```suggestion
            if (ASR::is_a<ASR::Dict_t>(*ASRUtils::type_get_past_const(type))) {
```
Does this simply the above duplication?

Reply:
This is the case of a normal dictionary. The logic for `Const dict` is copied from here. I thought about creating a function, but then did otherwise thinking it to be not a good choice.